eureka2-server/src/main/java/com/netflix/eureka2/server/audit/AuditServiceController.java

```diff
@@ -42,13 +43,13 @@
 
     private final SourcedEurekaRegistry<InstanceInfo> registry;
     private final AuditService auditService;
-    private final SelfInfoResolver serverIdentity;
+    private final Provider<SelfInfoResolver> serverIdentity;
 
     @Inject
     public AuditServiceController(
             SourcedEurekaRegistry registry,
             AuditService auditService,
-            SelfInfoResolver serverIdentity) {
+            Provider<SelfInfoResolver> serverIdentity) {
```

Comment:
We have a circular injection issues between the resolver, the TcpHanders and the TcpServers which necessitates the providers.